### PR TITLE
hashes/sha2{24,56}: Remove static variables from sha256

### PR DESCRIPTION
**Pull request by bergzand**
### Contribution description

This removes the static (thread-unsafe) variables from sha256 and hmac_sha256 to remove a potential footgun. The static variable is only used when the caller does not supply a pointer to store the digest and it is returned via the (undocumented) return value.

This commit removes this option and makes the digest argument mandatory.

### Testing procedure

The tests-hashes unittest should still work:
```
main(): This is RIOT! (Version: 2024.01-devel-240-gc98f3-pr/sha256/no_static)
Help: Press s to start test, r to print it is ready
s
START
...........................................
OK (43 tests)
```

### Issues/PRs references

Introduced in #124

### CODE REVIEW

File: sys/include/hashes/sha256.h [hunk 1]
```diff
@@ -132,7 +132,7 @@ static inline void sha256_final(sha256_context_t *ctx, void *digest)
  *                    be SHA256_DIGEST_LENGTH
  *                    if digest == NULL, one static buffer is used
```
[the remark this answers is not in the document]
Comment (bergzand) [reply sys/include/hashes/sha256.h:133]: Thanks! I just did a quick scan for a doxygen `@return` and nothing showed up.

**Comment by bergzand** [comment]
And removed the return argument from `hmac_sha256` also

**Review by mguetschow — APPROVED** [review]
LGTM then. Since this is an API breaking change, do we need to note that somewhere for the next release notes?

**Review by Teufelchen1 — APPROVED** [review]

**Review by maribu — APPROVED** [review]

**Comment by maribu** [comment]
Please squash

**Comment by OlegHahm** [comment]
I assume you have checked that no existing code calls the functions with passing a NULL pointer for `digest`?

**Comment by maribu** [comment]
> I assume you have checked that no existing code calls the functions with passing a NULL pointer for `digest`?

That code would need to make use of the return value, which by changing the signature will result in a compilation error that Murdock should catch.

**Comment by OlegHahm** [comment]
> > I assume you have checked that no existing code calls the functions with passing a NULL pointer for `digest`?
> 
> That code would need to make use of the return value, which by changing the signature will result in a compilation error that Murdock should catch.

Valid point. 😊

**Comment by mguetschow** [comment]
ping @bergzand for squashing

**Comment by bergzand** [comment]
> ping @bergzand for squashing

Done, sorry for the delay!

**Pull request merged**